src/plugins/discover/public/components/view_mode_toggle/view_mode_toggle.tsx

```diff
@@ -46,10 +46,16 @@ export const DocumentViewModeToggle = ({
     () => isLegacyTableEnabled({ uiSettings, isEsqlMode }),
     [uiSettings, isEsqlMode]
   );
+
   const [showPatternAnalysisTab, setShowPatternAnalysisTab] = useState<boolean | null>(null);
   const showFieldStatisticsTab = useMemo(
-    () => uiSettings.get(SHOW_FIELD_STATISTICS) && dataVisualizerService !== undefined,
-    [dataVisualizerService, uiSettings]
+    () =>
+      // If user opens saved search with field stats in ES|QL,
+      // we show the toggle with the mode disabled so user can switch to document view
+      // instead of auto-directing
```

Comment:
> instead of auto-directing

Is not what we do on L109-L113?

Reply:
For the most part yes if user opening a bookmarked link from the browser with Field stats previously set, it will redirect automatically. But when opening a saved search that a user previously saved, it doesn't redirect automatically. I think it makes sense because if a user is expecting field statistics from a saved search and the page is auto directing without any messaging that would be confusing.


https://github.com/user-attachments/assets/fc6aba79-9324-4937-9ebd-366d63842932



https://github.com/user-attachments/assets/5b14c4ce-56be-4644-aa25-bdedf385e105